nss/aad_auth/testdata/golden/TestGetEnt/try_to_list_group_with_empty_cache

```diff
@@ -0,0 +1,2 @@
+|
```

Comment:
with empty cache cases this should return the numeric equivalent of:

status: NSS_STATUS_NOTFOUND
errno: SUCCESS

(https://www.gnu.org/software/libc/manual/html_node/NSS-Modules-Interface.html).
(Note: the bug could be in the `internal/nss/` layer)

Reply:
The problem is in internal/nss indeed. The error that is return is nss.ErrNotFoundENoEnt, but according to the documentation, it should be nss.ErrNotFoundSuccess.

Reply:
I've managed to refactor the getAllEntries func to behave as expected, as the refactoring of the internal/nss functions would take more effort and we should probably discuss it in a peer session.
Also, the function errToCStatus might need a little review as well, since the values (status / errno) returned for ErrNotFoundSuccess are STATUS_SUCCESS / ENOENT rather than NOTFOUND / SUCCESS. (this is also included in the discussion session as well).

Reply:
I agree that this is the best direction for now to go on the empty tests.

well, we return the status before the errno? (which doesn’t map with the go error we return)

Reply:
I think so. So if I just invert the values it will be fine? In [here](https://github.com/ubuntu/aad-auth/blob/f98df3993b8619c5cad4bdfb4a0b5d38e568d3c8/nss/integration_tests/util_c.go#L38), I could change to be like:
```Go
nssStatus = C.ENOENT
errno = C.NSS_STATUS_SUCCESS
```

Reply:
No no no. What I meant is that in the Go error, we are quite logic by assessing errno, then status.

However, in C, you check the status code and then the errno. So inversing the returned value as the code does it logical from a C standpoint.